### PR TITLE
resource_manager, storage: create new storage endpoint for resource manager

### PR DESCRIPTION
**Pull request by CabinfeverB**
Signed-off-by: Cabinfever_B <cabinfeveroier@gmail.com>

### What problem does this PR solve?

Issue Number: Ref #5851 
This PR and #5886 will fix #https://github.com/pingcap/tiflow/issues/8110

### What is changed and how does it work?
Create a new storage endpoint and uses separate root path different from PD server

```commit-message
```

### Check List

Tests

- Unit test
- Integration test

Code changes

- Has persistent data change

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by HuSharp — APPROVED** [review]
LGTM!

**Review by JmPotato — APPROVED** [review]

File: server/grpc_service.go [hunk 1]
```diff
@@ -1888,9 +1888,9 @@ func (s *GrpcServer) StoreGlobalConfig(_ context.Context, request *pdpb.StoreGlo
 		name := item.GetName()
 		switch item.GetKind() {
 		case pdpb.EventType_PUT:
-			ops[i] = clientv3.OpPut(s.GetFinalPathWithinPD(request.GetConfigPath()+name), item.GetValue())
+			ops[i] = clientv3.OpPut(request.GetConfigPath()+name, item.GetValue())
```
Comment (binshi-bing) [inline server/grpc_service.go:1891]: nit: this line, compared to line 1893, has different coding style

Comment (binshi-bing) [reply server/grpc_service.go:1891]: "...)+name vs. "...) + name"

File: pkg/storage/endpoint/key_path.go [hunk 2]
```diff
@@ -43,6 +42,8 @@ const (
 	keyspaceIDInfix            = "id"
 	keyspaceAllocID            = "alloc_id"
 	regionPathPrefix           = "raft/r"
+	// resource group storage endpoint has prefix `resource_group`
+	resourceGroupSettingsPath = "settings"
```
Comment (binshi-bing) [inline pkg/storage/endpoint/key_path.go:46]: nit: alignment problem around "=".

**Review by binshi-bing — APPROVED** [review]
Left few trivial comments.
LGTM.

File: pkg/storage/endpoint/key_path.go (hunk 2 above)
Comment (JmPotato) [reply pkg/storage/endpoint/key_path.go:46]: According to the Go code formatter, it's separated by the comment above, so there is no need to align it with other equality signs.

File: server/grpc_service.go (hunk 1 above)
Comment (CabinfeverB) [reply server/grpc_service.go:1891]: This style is generated by go fmt

**Review by nolouch — APPROVED** [review]

**Pull request merged**